# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/michiel/blackbox/gcc34.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/Timer.hh.orig	2004-05-14 11:14:47.822257792 +0200
+++ ./src/Timer.hh	2004-05-14 11:16:53.078215992 +0200
@@ -98,9 +98,9 @@
   ~_timer_queue(void) {}
 
   void release(const _Tp& value) {
-    c.erase(std::remove(c.begin(), c.end(), value), c.end());
+    this->c.erase(std::remove(this->c.begin(), this->c.end(), value), this->c.end());
     // after removing the item we need to make the heap again
-    std::make_heap(c.begin(), c.end(), comp);
+    std::make_heap(this->c.begin(), this->c.end(), this->comp);
   }
   bool empty(void) const { return _Base::empty(); }
   size_t size(void) const { return _Base::size(); }
